src/backend/crew_ai/optimization/smart_keyword_provider.py

```
@@ -172,31 +200,65 @@ def _get_learning_hints(self, agent_role: str, user_query: str,
-                for hint in nl_hints:
-                    candidates.append({"text": hint, "priority": "high"})
+            cache_key = (user_query, safe_url)
+            if cache_key in self._nl_hints_cache:
+                nl_texts, nl_ids = self._nl_hints_cache[cache_key]
+            else:
+                nl_texts, nl_ids = self._get_nl_feedback_engine().get_hints_with_ids(
+                    user_query, safe_url, agent_role
+                )
+                self._nl_hints_cache[cache_key] = (nl_texts, nl_ids)
+            if nl_texts:
+                for text, nid in zip(nl_texts, nl_ids):
```

Ruff (0.15.14)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)